notmuch: use query as name for vfolder-from-query

Austin-Ray opened the pull request:
To provide a consistent experience for those who want human-readable
names, set the 'vfolder-from-query' mailbox's name to the user's query.

The previous behavior for 'vfolder-from-query' delegated mailbox
creation to lower-level backend-agnostic code, which lacks the ability
to set a name. As such, 'vfolder-from-query' mailboxes displayed the
encoded notmuch URI when the user wanted a human readable name.

Since we know a Mailbox will be always created from a valid
'vfolder-from-query' execution, we can create it at a high-level and
pass it to the lower-level components, which obviates the need for the
lower-level component to create it.

Closes #2033

Austin-Ray commented:
Amended commit fixed error in commit message. Accidentally said "Fixes" instead of "Closes".

flatcap commented on index.c:
```diff
@@ -2285,18 +2285,28 @@ int mutt_index_menu(struct MuttWindow *dlg)
+
+        if (nm_uri_from_query(NULL, buf, sizeof(buf)))
+        {
+          struct Mailbox *temp_mail = mx_path_resolve(buf);
```
(still unwritten) rule: Mailboxes are called `m` or `m_*`

flatcap commented on index.c:
```diff
@@ -2285,18 +2285,28 @@ int mutt_index_menu(struct MuttWindow *dlg)
+        if (nm_uri_from_query(NULL, buf, sizeof(buf)))
+        {
+          struct Mailbox *temp_mail = mx_path_resolve(buf);
+          temp_mail->name = mutt_str_strdup(query_unencoded);
```
We could make this a little more efficient:

```suggestion
          temp_mail->name = query_unencoded;
          query_unencoded = NULL;
```

flatcap commented:
> Accidentally said "Fixes" instead of "Closes".

Both are fine.  [GitHub's fairly flexible](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords).

Austin-Ray commented:
Addressed code review comments, and added comments for clarity. PR should be good to squash and merge. 

> Both are fine. [GitHub's fairly flexible](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords).

I'm just pedantic and want to properly reflect that this commit isn't a bug fix.

flatcap commented on index.c:
```diff
@@ -2285,18 +2285,31 @@ int mutt_index_menu(struct MuttWindow *dlg)
         else
-          main_change_folder(menu, op, NULL, buf, sizeof(buf), &oldcount, &index_hint, NULL);
-        break;
+          mutt_message(_("Failed to create query, aborting"));
```
Error path leaks `query_unencoded`

flatcap commented on index.c:
```diff
@@ -2285,18 +2285,31 @@ int mutt_index_menu(struct MuttWindow *dlg)
+
+        if (nm_uri_from_query(NULL, buf, sizeof(buf)))
+        {
+          // Create mailbox and set name. 
```
Trailing whitespace!
